link/src/main/java/com/stripe/android/link/ui/paymentmethod/PaymentMethodViewModel.kt

```diff
@@ -202,15 +204,33 @@ internal class PaymentMethodViewModel @Inject constructor(
         navigator.cancel(reason = PayAnotherWay)
     }
 
-    private fun completePayment(linkPaymentDetails: LinkPaymentDetails) {
-        val factory = ConfirmStripeIntentParamsFactory.createFactory(
-            clientSecret = requireNotNull(stripeIntent.clientSecret),
-            shipping = args.shippingValues?.toConfirmPaymentIntentShipping(),
+    private suspend fun completePayment(linkPaymentDetails: LinkPaymentDetails) {
+        val nextStep = intentConfirmationInterceptor.intercept(
+            clientSecret = stripeIntent.clientSecret,
+            paymentMethodCreateParams = linkPaymentDetails.paymentMethodCreateParams,
+            shippingValues = args.shippingValues?.toConfirmPaymentIntentShipping(),
+            setupForFutureUsage = null,
         )
 
-        val params = factory.create(linkPaymentDetails.paymentMethodCreateParams)
+        when (nextStep) {
+            is IntentConfirmationInterceptor.NextStep.Confirm -> {
+                confirmStripeIntent(nextStep.confirmParams)
+            }
+            is IntentConfirmationInterceptor.NextStep.HandleNextAction -> {
+                // This can't happen for Link
+            }
+            is IntentConfirmationInterceptor.NextStep.Fail -> {
+                val error = IllegalStateException(nextStep.message)
+                onError(error)
```

Comment:
Just confirming that this error never gets passed to the integrator. We only show the error message from this exception, right?

Reply:
Yes, this is displayed in the Link UI.